phfs.c

```diff
@@ -13,61 +13,388 @@
  * %LICENSE%
  */
 
-#include "errors.h"
+#include "phfs.h"
+#include "devs.h"
 #include "hal.h"
+
+#include "phoenixd.h"
 #include "plostd.h"
-#include "phfs.h"
+#include "errors.h"
+
+
+#define SIZE_PHFS_HANDLERS    8
+#define SIZE_PHFS_FILES       10
+
+
+enum { phfs_prot_raw = 0, phfs_prot_phoenixd };
 
-/* Configuration file for specific platform */
-#include "config.h"
 
+typedef struct {
+	char alias[8];
+	unsigned int major;
+	unsigned int minor;
+	unsigned int prot;
 
-phfs_handler_t phfs_handlers[PDN_NB];
+	dev_handler_t *h;
+} phfs_handler_t;
 
 
-handle_t phfs_open(u16 pdn, const char *name, u32 flags)
+typedef struct {
+	char alias[32];
+	addr_t addr;
+	size_t size;
+} phfs_file_t;
+
+
+struct {
+	phfs_handler_t handlers[SIZE_PHFS_HANDLERS];
+	unsigned int hCnt;
+
+	phfs_file_t files[SIZE_PHFS_FILES];
+	unsigned int fCnt;
+} phfs_common;
+
+
+static const char *phfs_getProtName(unsigned int id)
 {
-	handle_t handle;
+	switch (id) {
+	case phfs_prot_raw:
+		return "raw";
+
+	case phfs_prot_phoenixd:
+		return "phoenixd";
 
-	if (pdn >= PDN_NB) {
-		handle.h = ERR_ARG;
-		return handle;
+	default:
+		return "";
 	}
+}
+
 
-	return phfs_handlers[pdn].open(phfs_handlers[pdn].dn, name, flags);
+static int phfs_setProt(phfs_handler_t *ph, const char *prot)
+{
+	size_t sz = plostd_strlen(prot);
+
+	if (prot == NULL || (sz == 3 && plostd_strncmp(prot, "raw", 3) == 0))
+		ph->prot = phfs_prot_raw;
+	else if (sz == 8 && plostd_strncmp(prot, "phoenixd", 8) == 0)
+		ph->prot = phfs_prot_phoenixd;
+	else
+		return ERR_ARG;
+
+	return ERR_NONE;
+}
+
+
+static int phfs_getHandlerId(const char *alias)
+{
+	int i;
+	phfs_handler_t *ph;
+	size_t sz, aliasSz = plostd_strlen(alias);
+
+	for (i = 0; i < phfs_common.hCnt; ++i) {
+		ph = &phfs_common.handlers[i];
+		sz = plostd_strlen(ph->alias);
+		if ((sz == aliasSz) && (plostd_strncmp(alias, ph->alias, sz) == 0))
+			return i;
+	}
```

Comment:
```suggestion

	for (i = 0; i < phfs_common.hCnt; ++i) {
		ph = &phfs_common.handlers[i];
		if (plostd_strcmp(alias, ph->alias) == 0)
			return i;
	}
```
Same as `phfs_setProt()`.

Reply:
Yes, you have right. Changed to `plostd_strcmp()`